src/ray/raylet/scheduling_queue.cc

```diff
@@ -80,30 +80,25 @@ bool SchedulingQueue::TaskQueue::AppendTask(const TaskID &task_id, const Task &t
   RAY_CHECK(task_map_.find(task_id) == task_map_.end());
   auto list_iterator = task_list_.insert(task_list_.end(), task);
   task_map_[task_id] = list_iterator;
-  return true;
-}
-
-bool SchedulingQueue::TaskQueue::RemoveTask(const TaskID &task_id) {
-  auto task_found_iterator = task_map_.find(task_id);
-  if (task_found_iterator == task_map_.end()) {
-    return false;
-  }
-
-  auto list_iterator = task_found_iterator->second;
-  task_map_.erase(task_found_iterator);
-  task_list_.erase(list_iterator);
+  // Resource bookkeeping
+  current_resource_load_.AddResources(task.GetTaskSpecification().GetRequiredResources());
   return true;
 }
 
 bool SchedulingQueue::TaskQueue::RemoveTask(const TaskID &task_id,
-                                            std::vector<Task> &removed_tasks) {
+                                            std::vector<Task> *removed_tasks) {
```

Comment:
Why this change? We should avoid using raw pointers. If the idea is to share code with the other `RemoveTask`, let's just pass in a temporary `std::vectorTask>` there.

Reply:
This is the generic rule for arguments but the specific one for out-arguments is pointers:

```
In fact it is a very strong convention in Google code that [...] output arguments are pointers.
```

The convention is clear and we decided to follow it (I know that in other parts of the code we use non-const references for this, and we should change that I think).

It also avoids creating the temporary vector in this case.